### PR TITLE
chore(team-roles): Refactor Access component

### PR DESCRIPTION
**Pull request by dashed**

### CODE REVIEW

File: static/app/components/acl/access.tsx [hunk 1]
```diff
@@ -60,16 +60,10 @@ function Access({
...
-  const {access: orgAccess} = organization || {access: []};
-  const {access: teamAccess} = team || {access: [] as Team['access']};
-  const {access: projAccess} = project || {access: [] as Project['access']};
-
-  const hasAccess =
-    !access ||
-    access.every(acc => orgAccess.includes(acc)) ||
-    access.every(acc => teamAccess?.includes(acc)) ||
-    access.every(acc => projAccess?.includes(acc));
+  const hasAccess = hasEveryAccess(access, {organization, team, project});
```
Comment (evanpurkhiser) [inline static/app/components/acl/access.tsx:66]: `hasEveryAccess` where's that from?

Comment (dashed) [reply static/app/components/acl/access.tsx:66]: @evanpurkhiser at the bottom.

File: static/app/components/acl/access.tsx [hunk 2]
```diff
@@ -60,16 +60,10 @@ function Access({
   organization,
 }: Props) {
   const config = useLegacyStore(ConfigStore);
+  team = team ?? undefined;
+  project = project ?? undefined;
```
Comment (cathteng) [inline static/app/components/acl/access.tsx:64]: do you need to do this if null/undefined evaluate to false?

Comment (cathteng) [reply static/app/components/acl/access.tsx:64]: and `hasEveryAccess` is just a bunch of boolean checks?

Comment (dashed) [reply static/app/components/acl/access.tsx:64]: > do you need to do this if null/undefined evaluate to false?

@cathteng The props, `team` and `project` may be `null`.  Note that `hasEveryAccess` doesn't access `null` values. That's why I'm coercing them to be `undefined` beforehand.

**Review by cathteng — APPROVED** [review]

**Pull request merged**